File: FixMyCar/FixMyCar.Model/Entities/OrderDetail.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FixMyCar.Model.Entities
{
    public class OrderDetail
    {
        public int Id { get; set; }
        [ForeignKey(nameof(Order))]
        public int OrderId { get; set; }
        public Order Order { get; set; }
        [ForeignKey(nameof(StoreItem))]
        public int StoreItemId { get; set; }
        public StoreItem StoreItem { get; set; }
        public int Quantity { get; set; }
        public double UnitPrice { get; set; }
        public double TotalItemsPrice { get; set; }
        public double TotalItemsPriceDiscounted { get; set; }
        public double Discount { get; set; }
    }
}

File: FixMyCar/FixMyCar.UI/fixmycar_car_parts_shop/lib/src/models/order_detail/order_detail.dart
import 'package:json_annotation/json_annotation.dart';

part 'order_detail.g.dart';

@JsonSerializable()
class OrderDetail {
  int id;
  int orderId;
  int storeItemId;
  String storeItemName;
  int quantity;
  double unitPrice;
  double totalItemsPrice;
  double totalItemsPriceDiscounted;
  double discount;

  OrderDetail(
      this.id,
      this.orderId,
      this.storeItemId,
      this.storeItemName,
      this.quantity,
      this.unitPrice,
      this.totalItemsPrice,
      this.totalItemsPriceDiscounted,
      this.discount
      );

  factory OrderDetail.fromJson(Map<String, dynamic> json) => _$OrderDetailFromJson(json);

  Map<String, dynamic> toJson() => _$OrderDetailToJson(this);
}
